api/src/main/java/org/apache/iceberg/io/PositionOutputStream.java

```diff
@@ -29,4 +29,15 @@ public abstract class PositionOutputStream extends OutputStream {
    * @throws IOException If the underlying stream throws IOException
    */
   public abstract long getPos() throws IOException;
+
+  /**
+   * Return the stored length of the output object. Can differ from the current position for
+   * encrypting streams, and other non-length-preserving streams.
+   *
+   * @return output storage object length in bytes
```

Comment:
I think this should be the current stored length in bytes, since it will be called during a write.

Reply:
Sounds good, I'll update the text